consensus/hotstuff/integration/instance_test.go

```diff
@@ -245,8 +257,7 @@ func NewInstance(t require.TestingT, options ...Option) *Instance {
 			proposal := model.ProposalFromFlow(header, parent.View)
 
 			// store locally and loop back to engine for processing
-			in.headers.Store(header.ID(), header)
-			in.queue <- proposal
+			in.ProcessBlock(proposal)
```

Comment:
I believe compliance engine delivers this using a separate goroutine, should we do the same?